lib/dotcom/utils/service_date_time.ex

```diff
@@ -0,0 +1,228 @@
+defmodule Dotcom.Utils.ServiceDateTime do
+  @moduledoc """
+  A collection of functions that helps to work with date_times with regard to service times.
+  Currently, we consider the most general case where service ends at 02:59:59 and starts at 03:00:00.
+
+  In the future, we aim to add route-specific service times.
+
+  Note that functions like `service_today?/1` and `service_this_week?/1` are not exlusive.
+  A date_time can have service today and service this week.
+  I.e., today is both today and this week.
+
+  However, the `service_range/1` function *is* exclusive.
+  Given a date_time in today's service range, it will return :today.
+  """
+
+  require Logger
+
+  use Timex
+
+  import Dotcom.Utils.DateTime, only: [coerce_ambiguous_time: 1, now: 0, timezone: 0]
+
+  alias Dotcom.Utils
+
+  @type service_range() :: :past | :today | :this_week | :next_week | :later
+
+  @service_rollover_time Application.compile_env!(:dotcom, :service_rollover_time)
+  @timezone timezone()
+
+  @doc """
+  Returns the time at which service rolls over from 'today' to 'tomorrow'.
+  """
+  def service_rollover_time(), do: @service_rollover_time
+
+  @doc """
+  Get the service date for the given date_time.
+  If the time is before 3am, we consider it to be the previous day.
```

Comment:
This is a good use case for doctests and putting some usage examples right here in the doc - that makes the docs a much more powerful reference for how to use this function, and what to expect out of a return value.

(Doctests, rather than just usage examples, because that way if the docs get out of date, the tests will yell at us)

Reply:
I'm a little confused - I think you marked this as resolved, but I missed what the resolution actually was?